cloudinit/net/cmdline.py

```diff
@@ -108,6 +136,9 @@ def _klibc_to_config_entry(content, mac_addrs=None):
     note that IPV6PROTO is also written by newer code to address the
     possibility of both ipv4 and ipv6 getting addresses.
 
+    if DEVICE contains a '.' this indicates it is a VLAN device and
+    we must return config for the base device and vlan layer.
+
```

Comment:
No, no no. this is not true and will break all networking on s390x.

One cannot use a dot as an indicator or separator, as '.' is valid part vlan interface. There is no indication in ipconfig data if any part of the interface name indicates vlan or not.

One must introspect netlink / udevadm data to see if interface is vlan or not, and what the parent device is.